be/src/vec/exec/scan/vscan_node.cpp

```diff
@@ -771,6 +771,13 @@ Status VScanNode::_normalize_not_in_and_not_eq_predicate(VExpr* expr, VExprConte
                         ->get_function_state(FunctionContext::FRAGMENT_LOCAL));
         HybridSetBase::IteratorBase* iter = state->hybrid_set->begin();
         auto fn_name = std::string("");
+
+        // not in(any, null) == (x!=any and x!=null)
+        // (x!=null)=null=false, so not in(any, null) return always false
```

Comment:
`in (any, null)` does not always return false, in fact, it returns `null`.
an counter example:
`k not in (null)` returns null, so `not ( k not in (null)` returns null.
`select * from t where k not in (null) ` returns empty result set.
`select * from t where not (k not in (null)) ` returns empty result set too.